osu.Game.Rulesets.Taiko/UI/DefaultHitExplosion.cs

```diff
@@ -13,19 +14,25 @@
 
 namespace osu.Game.Rulesets.Taiko.UI
 {
-    internal class DefaultHitExplosion : CircularContainer
+    internal class DefaultHitExplosion : CircularContainer, IHitExplosion
     {
-        private readonly DrawableHitObject judgedObject;
+        public override bool RemoveWhenNotAlive => false;
```

Comment:
What's this one for?

Reply:
Truth be told, not sure myself now that I look at it, and everything seems to work without. Snipping...

Reply:
You've removed this override in taiko's legacy explosion, but not the one here (in the default explosion). Is there a reason for that?

Isn't the reason this was added because of `Expire(true)`, which would otherwise remove them from their parent, when the parent is the one being pooled?

Reply:
Pretty sure I got them both in f4e508b57051e887a00a8f4649e4616b762a8c8c...

The original reasoning was to make sure they don't get cleared from the pooled parent, sure. But it doesn't actually seem to have any actual difference, possibly because both these child components and the pooled parent share the sam lifetime. Not sure whether it's safe to rely on that, I'll double back on this change if you think that's best.

Reply:
Oh crap, on double checking I somehow got the entirely wrong override from mania. Yikes...

I think I'll revert the aforementioned commit wholesale. The rationale for the original change is as given above.

Reply:
> both these child components and the pooled parent share the same lifetime

Imo that's pretty dodgy to rely on. Make sense but... Anyway, other judgements (mania, osu) don't use `Expire(true)`, which is pretty unnecessary here too as you're using `LatestTransformEndTime`.

I've gone ahead and removed the `Expire(true)`s and overrides to match up.